### PR TITLE
feature airgap for downstream-cluster

### PR DESCRIPTION
**Pull request by alknopfler**
- add ATIP telco examples to deploy a downstream-cluster in an air-gap environment

### CODE REVIEW

**Comment by alknopfler** [comment]
Tested in the lab, using a airgap (isolated) network in libvirtd and:
- mgmt cluster air-gapped
- using capi to deploy the downstream cluster air-gapped as well.

File: telco-examples/edge-clusters/airgap/README.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  |-- install.sh
+  |-- rke2-images-cilium.linux-amd64.tar.zst
+  |-- rke2-images-core.linux-amd64.tar.zst
+  |-- rke2-images-multus.linux-amd64.tar.zst
+  |-- rke2-images.linux-amd64.tar.zst
+  |-- rke2.linux-amd64.tar.gz
+  `-- sha256sum-amd64.txt
+  ```
+
+2. Preload your registry with the helm-chart oci images required for the edge cluster.
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/airgap/README.md:55]:
```suggestion
2. Prepare your artifact registry.
```

File: telco-examples/edge-clusters/airgap/README.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  |-- rke2-images-cilium.linux-amd64.tar.zst
+  |-- rke2-images-core.linux-amd64.tar.zst
+  |-- rke2-images-multus.linux-amd64.tar.zst
+  |-- rke2-images.linux-amd64.tar.zst
+  |-- rke2.linux-amd64.tar.gz
+  `-- sha256sum-amd64.txt
+  ```
+
+2. Preload your registry with the helm-chart oci images required for the edge cluster. 
+  - You need to create a list with the helm-chart oci images required for the edge cluster. For example, for telco scenarios,  you can use the following list:
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/airgap/README.md:56]:
```suggestion
  - You need to create a list with the Helm charts required for the edge cluster. For example, for telco scenarios, you can use the following list:
```

File: telco-examples/edge-clusters/airgap/README.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    ``` 
+    $ cat > edge-release-helm-oci-artifacts.txt <<EOF
+    edge/sriov-network-operator-chart:1.2.2
+    edge/sriov-crd-chart:1.2.2
+    EOF
+    ```
+  - Using the [following script](https://github.com/suse-edge/fleet-examples/blob/release-3.0/scripts/day2/edge-save-oci-artefacts.sh) and the list created above, you can generate in local the tarball with the helm-chart oci images required for the edge cluster.
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/airgap/README.md:63]:
```suggestion
  - Using the [following script](https://github.com/suse-edge/fleet-examples/blob/release-3.0/scripts/day2/edge-save-oci-artefacts.sh) and the list created above, you can generate a tarball containing all necessary Helm charts locally.
```

File: telco-examples/edge-clusters/airgap/README.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    ```
+    $ ./edge-save-oci-artefacts.sh -al ./edge-release-helm-oci-artifacts.txt -s registry.suse.com
+    Pulled: registry.suse.com/edge/sriov-network-operator-chart:1.2.2
+    Pulled: registry.suse.com/edge/sriov-crd-chart:1.2.2
+    a edge-release-oci-tgz-20240705
+    a edge-release-oci-tgz-20240705/sriov-network-operator-chart-1.2.2.tgz
+    a edge-release-oci-tgz-20240705/sriov-crd-chart-1.2.2.tgz
+    ```
+  - Upload your tarball to your private registry to preload with the helm chart oci images downloaded in the previous step using the [following script](https://github.com/suse-edge/fleet-examples/blob/release-3.0/scripts/day2/edge-load-oci-artefacts.sh):
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/airgap/README.md:72]:
```suggestion
  - Upload your tarball to your private registry to load the downloaded Helm charts using the [following script](https://github.com/suse-edge/fleet-examples/blob/release-3.0/scripts/day2/edge-load-oci-artefacts.sh):
```

File: telco-examples/edge-clusters/airgap/README.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    ```
+    $ tar zxvf edge-release-oci-tgz-20240705.tgz
+    $ ./edge-load-oci-artefacts.sh -ad edge-release-oci-tgz-20240705 -r myregistry:5000
+    ```
+
+3. Preload your registry with the rest of the images as well as your workload images required for the edge cluster.
```
Comment (atanasdinov) [inline telco-examples/edge-clusters/airgap/README.md:78]:
```suggestion
3. Preload your registry with the necessary container images (incl. your workload ones) required for the edge cluster. 
```

**Review by atanasdinov — COMMENTED** [review]
LGTM. I only have wording suggestions in order to simplify the descriptions, as well as dropping some repetitions here and there.

File: telco-examples/edge-clusters/airgap/README.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+### Prerequisites
+
+To prepare an airgap environment, we will use a private local registry (already deployed and configured) to store some images (helm-chart oci images and some other images), and we will use the Edge Image Builder to include the rke2-images artifacts which are a requirement to use the cluster-api provider for rke2.
+
+Using the example folder `telco-examples/edge-clusters/airgap/eib` we will create the basic structure in order to build the image for the edge cluster: 
+
+You need to modify the following values in the `telco-edge-airgap-cluster.yaml` file:
+
+- `${ROOT_PASSWORD}` - The root password for the management cluster. This could be generated using `openssl passwd -6 PASSWORD` and replacing PASSWORD with the desired password, and then replacing the value in the `telco-edge-cluster.yaml` file.
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/README.md:28]: ohh good catch!!! copy&paste from mgmt-cluster. Thanks

File: telco-examples/edge-clusters/airgap/README.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+- `${SCC_REGISTRATION_CODE}` - The registration code for the SUSE Customer Center for the SLE Micro product. This could be obtained from the SUSE Customer Center and replacing the value in the `telco-edge-cluster.yaml` file.
+
+You need to modify the following folder:
+
+- `base-images` - To include inside the `SLE-Micro.x86_64-5.5.0-Default-RT-GM.raw` image downloaded from the SUSE Customer Center.
+
+### preparing the airgap aritfacts
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/README.md:35]: also another typo in artifacts word. Thanks

File: telco-examples/edge-clusters/airgap/README.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+$ kubectl apply -f bmh-example.yaml
+```
+
+The new Baremetal host will be enrolled changing the state from registering to inspecting and available. You could check the status using the following command:
+
+``` 
+$ kubectl get bmh
+```
+
+#### Provision the new host using the CAPI manifests and the image generated in an air-gap environment
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/README.md:170]: let's change to: Provision the new host using the CAPI manifests

File: telco-examples/edge-clusters/airgap/README.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+
+### Building the Edge Cluster Image using EIB
+
+All the following commands in this section could be executed using any linux laptop/server x86_64 with podman installed. You don't need to have a specific environment to build the image.
+
+#### Generate the image with our configuration for Telco profile
+
+```
+$ cd telco-examples/edge-clusters
+$ sudo podman run --rm --privileged -it -v $PWD:/eib \
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/README.md:122]: depending on the cgroup version, right?

File: telco-examples/edge-clusters/airgap/README.md (hunk 1 above)
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/README.md:55]: mmm I'd like to keep the preload. it's really common when you're working with disconnected environments

File: telco-examples/edge-clusters/airgap/README.md (hunk 3 above)
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/README.md:63]: much better. Thanks

File: telco-examples/edge-clusters/airgap/README.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+#### Provision the new host using the CAPI manifests and the image generated in an air-gap environment
+
+Once the new Baremetal host is available, you need to provision the new host using the CAPI manifests and the image generated in the previous step.
+
+The first thing is to modify the `telco-capi-airgap.yaml` file and replace the following with your values:
+
+- `${TLS_BASE64_CERT}` and `${TLS_BASE64_KEY}` - In the case your private registry contains a client certificate and key, you can include it and need to be encoded them in base64 and replace the values in the `telco-capi-airgap.yaml` file.
+- `${CA_BASE64_CERT}` - In the case your private registry enables TLS, at least you need to include the CA certificate to be used for the private registry. You need to encode it in base64 and replace the value in the `telco-capi-airgap.yaml` file.
+- `${REGISTRY_USERNAME}` and `${REGISTRY_PASSWORD}` - In the case your private registry enables the basic-auth mecanism, you need to include the username and password to be used for the private registry. You need to encode it in base64 and replace the values in the `telco-capi-airgap.yaml` file.
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply telco-examples/edge-clusters/airgap/README.md]: it's ok, Base64 encoded

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**